stores/wallet.ts

```diff
@@ -88,14 +164,263 @@ export const useWalletStore = defineStore('wallet', () => {
+  }) {
+    const shouldMigrate = !boundEVMWallet && !!boundLikeWallet
+    return {
+      statusCode: 401,
+      data: {
+        level: 'warning',
+        title: shouldMigrate
+          ? $t('account_register_error_email_already_used_migrate_title')
+          : $t('account_register_error_email_already_used_by_wallet_title'),
+        description: getEmailAlreadyUsedErrorMessage({
+          email,
+          evmWallet: boundEVMWallet,
+          likeWallet: boundLikeWallet
+        }),
+        tags: [
+          { label: loginMethod, icon: 'i-material-symbols-login-rounded', class: 'font-mono' },
+          { label: walletAddress, icon: 'i-material-symbols-key-outline-rounded', class: 'font-mono' }
+        ],
+        actions: shouldMigrate
+          ? [{
+              label: $t('account_register_error_contact_support'),
+              color: 'secondary',
+              variant: 'subtle',
+              onClick: async () => {
+                await navigateTo(getLikeCoinV3BookMigrationSiteURL.value({ utmSource: 'login_email_already_used' }), {
+                  external: true,
+                  open: { target: '_blank' }
+                })
+              }
+            }]
+          : []
+      }
+    }
+  }
+
+  function getEmailAlreadyUsedErrorMessage ({
+    email,
+    evmWallet,
+    likeWallet
+  }: {
+    email: string
+    evmWallet?: string
+    likeWallet?: string
+  }) {
+    if (evmWallet) {
+      return $t('account_register_error_email_already_used_with_evm_wallet', { email, evmWallet })
+    }
+    if (likeWallet) {
+      return $t('account_register_error_email_already_used_with_like_wallet', { email, likeWallet })
+    }
+    return $t('account_register_error_email_already_used', { email })
+  }
+
+  async function register ({
+    walletAddress,
+    email: prefilledEmail,
+    loginMethod,
+    magicUserId,
+    magicDIDToken
+  }: {
+    walletAddress: string
+    email?: string
+    loginMethod: string
+    magicUserId?: string
+    magicDIDToken?: string
+  }) {
+    let tempAccountId = generateAccountIdFromWalletAddress(walletAddress)
+    try {
+      await usePostRegisterCheck({ accountId: tempAccountId })
+    } catch (error) {
+      if (error instanceof FetchError && error.data?.error === 'USER_ALREADY_EXIST') {
+        tempAccountId = error.data.alternative as string
+      } else {
+        throw error
+      }
+    }
+
+    const payload = {
+      accountId: tempAccountId,
+      email: prefilledEmail
+    }
+
+    if (!payload.email) {
+      type ModalResult = { accountId: string; email: string; displayName?: string }
+      let modalResult: ModalResult | null = null
+      // Close login panel first to avoid focus trap
+      bookstoreApiStore.closeLoginPanel()
+      await nextTick()
+      await new Promise<void>((resolve) => {
+        modal.open(RegistrationModal, {
+          email: payload?.email,
+          accountId: payload?.accountId,
+          isAccountIdHidden: true,
```

Comment:
Consider if we want to show `accountId` or any extra handling if it is a retry attempt, we can control this by pass extra params when calling `register`